rdrf/rdrf/forms/form_title_helper.py

```diff
@@ -0,0 +1,31 @@
+from rdrf.models.definition.models import FormTitle
+
+
+class FormTitleHelper:
+
+    def __init__(self, registry, default_name):
+        self.registry = registry
+        self.default_name = default_name
+
+    def title_for_user(self, user):
+        entry = FormTitle.objects.filter(
+            registry=self.registry,
+            group__in=user.groups.all(),
+            default_name=self.default_name
+        ).order_by('order').first()
+        return entry.custom_name if entry else self.default_name
+
+    def title_dict_for_user(self, user):
+        results = FormTitle.objects.filter(
+            registry=self.registry,
+            group__in=user.groups.all(),
+        ).order_by('default_name', 'order')
```

Comment:
You could extract the querysets from the 2 methods to make it more DRY.
Also, getting the mappings out could be simplified IMO:

```
mappings = {x[0]: x[0] for x in FormTitle.FORM_TITLE_CHOICES}
for r in results:
    mappings[r.default_name] = r.custom_name
```

Reply:
That won't help much as I want the mappings set with the default choices only if there's no current mapping for that default name.